prusti/src/bin/prusti-driver.rs

```diff
@@ -57,6 +57,7 @@ fn main() {
     args.push("-Zdump-mir-dir=log/mir/".to_owned());
     args.push("-Zdump-mir=renumber".to_owned());
     args.push("-Zalways-encode-mir".to_owned());
+    args.push("-Zcrate-attr=feature(stmt_expr_attributes)".to_owned());
```

Comment:
Why do you need this? Relying on unstable features can cause serious maintenance problems.

Reply:
> Why do you need this? Relying on unstable features can cause serious maintenance problems.

Well, so can linking against the nightly compiler. But you see, I have to annotate closures with specifications, and closures are expressions, hence the feature that allows me to add annotations to expressions. Specifically, I'm currently using the feature like this when rewriting the `closure!()` macro:
```Rust
#[prusti::post_spec_id_ref = "3fa334f2e58746f98899c3e81532413f"]
|a: i32, b: i32| { ... }
```
But if this is a problem, feel free to suggest an alternative solution!

Reply:
The specifications are also desugared into closures and adding attributes on them does not seem to cause problems. Maybe, something like this would work:
```rust
{
  #[prusti::post_spec_id_ref = "3fa334f2e58746f98899c3e81532413f"]
  let cl = |a: i32, b: i32| { ... };
  cl
}
```

Reply:
> The specifications are also desugared into closures and adding attributes on them does not seem to cause problems. Maybe, something like this would work:
> 
> ```rust
> {
>   #[prusti::post_spec_id_ref = "3fa334f2e58746f98899c3e81532413f"]
>   let cl = |a: i32, b: i32| { ... };
>   cl
> }
> ```

Unfortunately not:
```
thread 'rustc' panicked at 'not implemented: Closure(DefId(0:7 ~ closure_true[317d]::test[0]::{
{closure}}[0]), [i8, extern "rust-call" fn((i32, i32)) -> i32, ()])', prusti-viper/src/encoder/
procedure_encoder.rs:4719:22
```
Seems like it has something to do with the assignment (which we're going to have to implement at some point, anyways, but we're not there yet).

Reply:
> which we're going to have to implement at some point, anyways, but we're not there yet

Correct me if I am wrong, but you will have to implement assignment support before you can do anything with closures because the Rust compiler always emits temporary variables and many assignments.

Reply:
> Correct me if I am wrong, but you will have to implement assignment support before you can do anything with closures because the Rust compiler always emits temporary variables and many assignments.

Yes, but so far, I have not yet succeeded in verifying the closure _bodies_, which I decided would be my first goal. And for that, you don't need assignments (unless you have higher-order closures or something).

Reply:
Then again, it doesn't work currently anyway, so I dropped the experimental feature for this PR in my latest commit.